encoder.go

```diff
@@ -0,0 +1,414 @@
+package rdb
+
+import (
+	"fmt"
+	"math"
+	"sync/atomic"
+	"time"
+)
+
+type Encoder struct {
+	writer       RDBWriter
+	countPos     int64
+	count        int64
+	countWithExp int64
+	backlenBuf   []byte
+	version      string
+	begin        atomic.Bool
+}
+
+func NewFileEncoder(path string, version string) (*Encoder, error) {
```

Comment:
minor: maybe we can rename this to redis version? i thought this was RDB version at first